### ISSUE TITLE
Take fileinfo kwarg in add_target and add_target_to_bin methods

### ISSUE THREAD

**Issue by lukpueh** [theupdateframework/tuf#1003]
**Description of issue or feature request**:

The repository tool `add_target` and `add_target_to_bin` methods can be used to add target **paths** to the targets metadata in memory. The corresponding target file hashes are produced later e.g. on `write` or `writeall`.

It should be possible to add target paths using those functions, along with target file hashes that are generated out of band (this is required for PEP 458).

**Current behavior**:
`add_target` and `add_target_to_bin` don't take a `fileinfo` argument.

*Implementation hint*: `roledb` currently has a paths dictionary, where target paths are listed as dict keys and the optionally passed `custom` dict as value:

``` python
{"paths" : {"<TARGET PATH>": "custom": <CUSTOM DICT>}, ...}}
```

**Expected behavior**:

`add_target` and `add_target_to_bin` take an optional `fileinfo` argument that adheres to `tuf.formats.FILEINFO_SCHEMA`.

*Implementation hint*:  probably requires a change to `roledb` to store all `fileinfo` properties, e.g.:
``` python
{
"paths" : {
  "<TARGET PATH>": {
    "length": <LENGTH>,
    "hashes": <HASHES>,
    "version": <VERSION>,
    "custom": <CUSTOM DICT>
    }, ... 
  }
}
```

**Discussion**:

- `fileinfo` as returned by `tuf.formats.make_fileinfo` optionally includes the `custom` field (see above). It feels redundant to have `add_target` accept both `custom` and `fileinfo`. If we are fine with an interface change, I suggest to replace the former with the latter. Otherwise, we can have both, and fail if someone passes both `custom` and `fileinfo`

- Functions that take multiple target paths as list of string,  i.e. `add_targets`, `delegate`, `delegate_hashed_bins`, IMO don't need to provide this feature.

**Comment by woodruffw** [comment]
To clarify this on the downstream side, this is *roughly* how I'd expect to consume this API (the `utils` calls are likely to change, that's just me scratchpadding):

```python
from warehouse.tuf import utils

@task(bind=True, ignore_result=True, acks_late=True)
def add_target(task, request, file):
    repository = utils.open_repository(request)
    fileinfo = utils.make_fileinfo(file)
    
    repository.add_target_to_bin(file.path, fileinfo=fileinfo)
    # signing, writing operations
```

**Comment by lukpueh** [comment]
Fixed in #1007
